src/org/ohdsi/usagi/ui/ShowReviewStatsDialog.java

```diff
@@ -0,0 +1,165 @@
+/*******************************************************************************
+ * Copyright 2021 Observational Health Data Sciences and Informatics & The Hyve
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+package org.ohdsi.usagi.ui;
+
+import org.ohdsi.usagi.CodeMapping;
+import org.ohdsi.utilities.collections.Pair;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShowReviewStatsDialog extends JDialog {
+
+	private static final long	serialVersionUID	= 2028328868610404663L;
+
+	public ShowReviewStatsDialog() {
+		Font headerFont = new Font("Arial", Font.BOLD,12);
+
+		java.util.List<CodeMapping> selectedCodeMappings = Global.mappingTablePanel.getSelectedCodeMappings();
+		List<CodeMapping> codeMappings;
+		if (selectedCodeMappings.size() > 1) {
+			codeMappings = selectedCodeMappings;
+		} else {
+			codeMappings = Global.mapping;
+		}
+
+		setTitle("Review statistics");
+		setLayout(new GridBagLayout());
+
+		GridBagConstraints g = new GridBagConstraints();
+		g.fill = GridBagConstraints.BOTH;
+		g.ipadx = 50;
+		g.ipady = 10;
+
+		g.gridx = 0;
+		g.gridy = 0;
+		add(new JLabel(String.format("Number of (selected) source codes: %d", codeMappings.size())), g);
+
+		// Mapping status
+		g.gridx = 0;
+		g.gridy++;
+		JLabel l = new JLabel("By mapping status:");
+		l.setFont(headerFont);
+		add(l, g);
+
+		Map<CodeMapping.MappingStatus, Integer> mappingStats = new HashMap<>();
+		for (CodeMapping codeMapping : codeMappings) {
+			mappingStats.putIfAbsent(codeMapping.getMappingStatus(), 0);
+			Integer c = mappingStats.get(codeMapping.getMappingStatus());
+			mappingStats.put(codeMapping.getMappingStatus(), ++c);
+		}
+
+		mappingStats.forEach((key, value) -> {
+			g.gridx = 0;
+			g.gridy++;
+			add(new JLabel(String.format("%s - %d", key, value)), g);
```

Comment:
You're repeating this same sequence quite a bit in this method. You could think using a method like

```java
JLabel addLabel(GridBagConstraints g, String message) {
    g.gridx = 0;
    g.gridy++;
    JLabel label = new JLabel(message);
    add(label, g);
    return label;
}
```